lazy_tensor_core/lazy_tensor_core/core/extract_compiled_graph.py

```diff
@@ -90,9 +91,29 @@ def tolazydevice(dev):
             return torch.device("lazy", index=dev.index)
         return dev
 
+    def hasDeviceArg(args, kwargs):
+        return any(isinstance(arg, torch.device) for arg in itertools.chain(args, kwargs.values()))
+
     for nd in model.graph.nodes:
         nd.args = tuple(tolazydevice(arg) for arg in nd.args)
         nd.kwargs = {k: tolazydevice(v) for k, v in nd.kwargs.items()}
+
+        # For torchbench like yolov3, hf_Bart, dynamo generates Fx graph that return
+        # eager tensors on the default device
+        # (check https://gist.github.com/shunting314/eabdf6c769c59bc384469717b8f9bb7f for yolove,
+        # and https://gist.github.com/shunting314/8d5e2d9348a3258959d3954186c48814 for hf_Bart).
+        # To force those tensors on the lazy device, we can not simply override
+        # the device argument since there is no explicit device argument.
+        # What we are doing here is, for the list of covered tensor factory methods
+        # we add a lazy device argument explicity.
+        #
+        # TODO: This solution is no ideal since we may miss some factory methods. In future
+        # when we support lazy mode, this method can be replaced by that.
+        if nd.target in (torch.tensor, torch.full, torch.arange) and not hasDeviceArg(nd.args, nd.kwargs):
```

Comment:
I think this makes sense until we have lazy mode;

but would it be possible to get a more complete list of factory functions here?

Reply:
yep, I only added 3 since these are the minimal set to fix the models in torchbench. (Check the Fx graph paste I added in the comment above). Even if I try my best , I may still missing some factory methods (at least missing some methods that may be added in future). So I didn't do that. Hopefully we can replace this with lazy mode soon.